BlueDBOnDisk/src/main/java/org/bluedb/disk/collection/ReadableCollectionOnDisk.java

```diff
@@ -140,6 +140,11 @@ protected static SegmentSizeSetting determineSegmentSize(ReadableCollectionMetad
 
 	protected static Class<? extends BlueKey> determineKeyType(ReadableCollectionMetadata metaData, Class<? extends BlueKey> providedKeyType) throws BlueDbException {
 		Class<? extends BlueKey> storedKeyType = metaData.getKeyType();
+		
+		if(storedKeyType == null && providedKeyType == null) {
+			throw new BlueDbException("Unable to determine key type for collection at " + metaData.getPath() + ". You must initialize a pre-existing valid collection or provide a key type.");
+		} 
+		

```

Comment:
This takes care of causing the constructor to throw an exception if the collection is invalid due to not being able to determine a key type.